synchronization/src/main.c

```diff
@@ -115,6 +115,11 @@ int main(void)
 	}
 #endif
 
-	k_thread_start(&thread_a_data);
+    k_thread_start(&thread_a_data); 
+    k_thread_join(&thread_a_data, K_MSEC(400000));
+
+	__asm__("add a6, zero, 1\n"
```

Comment:
Just a minor nits: could you add a comment so that people know what is going on here? Something like "Call into Miralis to exit successfully"